Crypto/PubKey/RSA/PSS.hs

```diff
@@ -161,30 +163,37 @@ verifyDigest :: HashAlgorithm hash
              -> ByteString  -- ^ Signature
              -> Bool
 verifyDigest params pk digest s
-    | public_size pk /= B.length s        = False
+    | B.length s /= k                     = False
+    | B.any (/= 0) pre                    = False
     | B.last em /= pssTrailerField params = False
     | not (B.all (== 0) ps0)              = False
     | b1 /= B.singleton 1                 = False
-    | otherwise                           = h == B.convert h'
+    | otherwise                           = B.eq h h'
         where -- parameters
               hashLen   = hashDigestSize (pssHash params)
               mHash     = B.convert digest
-              dbLen     = public_size pk - hashLen - 1
-              pubBits   = public_size pk * 8 -- to change if public_size is converted in bytes
+              k         = public_size pk
+              emLen     = if emTruncate pubBits then k - 1 else k
```

Comment:
Why not just ```emLen = (pubBits - 1) `div` 8```? If we want a consistency sanity check between `pubBits` and `k` it could  be ```k == (pubBits + 7) `div` 8```, if there's ever any doubt that the `public_size` is consistent with the modulus.  Note that IIRC, unfortunately, the library does not hide the RSA key structure internals, and provides no constructors that ensure correctness of the provided values.  The application gets to set the modulus and `public_size` as it pleases.  This appears to be difficult to fix without a breaking change to the library, but should perhaps be done at some point, bumping the major number, and fixing all the dependencies in the X.509 and TLS libraries.